docs/codeactions/code-actions.md

```diff
@@ -0,0 +1,93 @@
+---
+id: codeactions
+title: Code Actions
+---
+
+
+Code Actions are used to provide automatic refactoring or rewriting of your current code in order to enhance its readability or save users' time.
+
+In Visual Studio Code, to use them you can click on the yellow light bulb appearing on your code, near the cursor and then, to pick the desired action to perform on the code. Alternatively, you can use the `editor.action.quickFix` shortcut.
+
+For other editors this should also be available via specified shortcut.
+
+Here is a list of the code actions supported by Metals. Some of these code actions support much more complex use cases, which you would eventually discover through your code, by clicking on the magic 💡, as it appears, to get your surprise!
```

Comment:
```suggestion
Here is a list of the code actions supported by Metals.
```
I would omit or rephrase the last sentence here :smile_cat:
This part adds some ambiguity: what means `much more complex use cases`, `you would eventually discover` - when, where?

Reply:
well, the current gifs are only covering the simplest cases for getting the message across. So let's say the use case of covering partial functions in map/flatMap is not shown, so the user would eventually discover this more complex one through clicking the light bulb.

Reply:
> So let's say the use case of covering partial functions in map/flatMap is not shown

From user viewpoint there is no difference. I think, people would be more surprised if it doesn't work in some cases rather than it support some particular tree inside an expression.

> so the user would eventually discover 

Yep, sometimes actions are discovered by a user but I think it's a rare occasion.

I think, in general, interaction with actions comes not from light-bulb suggestion.
User know which actions Metals supports and calls them to avoid manual rewriting.
So, you want to specify a type of your val - you know that there is an action. The same with for-comp conversion and others. Usually you know in advance that some particular code should have a rewrite action.

Reply:
ok, no problem.